pkg/virt-handler/vm.go

```diff
@@ -960,6 +980,59 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 	return nil
 }
 
+func _guestAgentCommandSubsetSupported(requiredCommands []string, commands []v1.GuestAgentCommandInfo) bool {
+	var found bool
+	for _, cmd := range requiredCommands {
+		found = false
+		for _, foundCmd := range commands {
+			if cmd == foundCmd.Name {
+				if foundCmd.Enabled {
+					found = true
+				}
+				break
+			}
+		}
+		if found == false {
+			return false
+		}
+	}
+	return true
+
+}
+
+func isGuestAgentSupported(vmi *v1.VirtualMachineInstance, commands []v1.GuestAgentCommandInfo) bool {
+	if !_guestAgentCommandSubsetSupported(RequiredGuestAgentCommands, commands) {
+		return false
+	}
+
+	checkSSH := false
+	checkPasswd := false
+
+	if vmi != nil && vmi.Spec.AccessCredentials != nil {
+		for _, accessCredential := range vmi.Spec.AccessCredentials {
+			if accessCredential.SSHPublicKey != nil {
+				// defer checking the command list so we only do that once
+				checkSSH = true
+			}
+			if accessCredential.UserPassword != nil {
+				// defer checking the command list so we only do that once
```

Comment:
we only support injecting user/password with qemu guest agent here, but there might be new propagation methods in the future. it would be worth verifing the propagation method is qemuguestagent here as well.

Reply:
Fair point! thanks for the reminder.